packages/core/test/util/initializedTooLateHeuristic_test.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+
+'use strict';
+
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const path = require('path');
+const requireHook = require('../../src/util/requireHook');
+const initializedTooLateHeurstic = require('../../src/util/initializedTooLateHeuristic');
+
+describe('[UNIT] util.initializedTooLateHeurstic', () => {
+  const instrumentedModules = [];
+
+  before(() => {
+    sinon.stub(requireHook, 'onFileLoad').callsFake(function fake(m) {
+      instrumentedModules.push(m);
+    });
+    sinon.stub(requireHook, 'onModuleLoad').callsFake(function fake(m) {
+      instrumentedModules.push(m);
+    });
+
+    require.cache['/@contrast/agent/'] = {};
+
+    const tracing = require('../../src/');
+    tracing.init();
+  });
+
+  after(() => {
+    delete require.cache['/@contrast/agent/'];
```

Comment:
_question:_ Do we also need to undo the `sinon.stub` calls on the `requireHook` module in `after` here`? I think we should do that.

Reply:
TODO: check if `sinon.reset` is needed